sympy/stats/rv.py

```diff
@@ -1076,57 +1079,96 @@ def sample(expr, condition=None, size=(), library='scipy',
+    [array([[ 0.07664441, -1.42557566,  3.66110395,  5.37576465],
+       [ 0.4993775 ,  7.80504228, -0.22978144,  4.5228755 ],
+       [ 5.91415701,  6.85960422,  9.52447392,  0.63699082]]),
+    array([[-1.89958054,  7.22563251,  1.74164548,  7.39832687],
+       [ 3.6053125 ,  4.83748511,  4.77375537,  0.35377489],
+       [ 5.31794739,  8.57816162,  2.16933622, -6.13675074]])]
+
     >>> G = Geometric('G', 0.5) # Discrete Random Variable
-    >>> samp_list = next(sample(G, size=3)) # doctest: +SKIP
+    >>> samp_list = sample(G, size=3)
     >>> samp_list # doctest: +SKIP
-    array([10,  4,  1])
-    >>> [sam in G.pspace.domain.set for sam in samp_list] # doctest: +SKIP
+    [1, 3, 2]
+    >>> [sam in G.pspace.domain.set for sam in samp_list]
     [True, True, True]
     >>> MN = Normal("MN", [3, 4], [[2, 1], [1, 2]]) # Joint Random Variable
-    >>> samp_list = next(sample(MN, size=4)) # doctest: +SKIP
+    >>> samp_list = sample(MN, size=4)
     >>> samp_list # doctest: +SKIP
-    array([[4.22564264, 3.23364418],
-           [3.41002011, 4.60090908],
-           [3.76151866, 4.77617143],
-           [4.71440865, 2.65714157]])
-    >>> [tuple(sam) in MN.pspace.domain.set for sam in samp_list] # doctest: +SKIP
+    [array([2.85768055, 3.38954165]),
+     array([4.11163337, 4.3176591 ]),
+     array([0.79115232, 1.63232916]),
+     array([4.01747268, 3.96716083])]
+    >>> [tuple(sam) in MN.pspace.domain.set for sam in samp_list]
     [True, True, True, True]
 
 
     Returns
     =======
 
-    sample: iterator object
-        iterator object containing the sample/samples of given expr
+    sample: float/list/numpy.ndarray
+        one sample or a collection of samples of the random expression
+
+        sample(X) --> returns float object
+        sample(X, numsamples=int) --> returns list object
+        sample(X, size=int/tuple) --> returns numpy.ndarray object
+
 
     """
-    ### TODO: Remove the user warnings in the future releases
-    message = ("The return type of sample has been changed to return an "
-                  "iterator object since version 1.7. For more information see "
-                  "https://github.com/sympy/sympy/issues/19061")
-    warnings.warn(filldedent(message))
-    return sample_iter(expr, condition, size=size, library=library,
+
+    iterator = sample_iter(expr, condition, size=size, library=library,
                                                         numsamples=numsamples, seed=seed)
 
+    if numsamples != 1:
+        SymPyDeprecationWarning(
+                 feature="numsamples parameter",
+                 issue=21563,
+                 deprecated_since_version="1.9").warn()
+
+        if library == 'pymc3':
+            raise NotImplementedError("numsamples is not currently implemented for pymc3")
+        return [next(iterator) for i in range(numsamples)]
+
+    if library == 'pymc3' and size == ():
+        return next(iterator)[0]
+
+    return next(iterator)
```

Comment:
Why does pymc3 need to be special cased here?

Can that not just be handled in the pymc3 code?

Does `sample_iter` with `size=()` return different results for pymc3 than the other implementations?

Reply:
yes, it does.
```python
>>> next(sample_iter(Normal('',10,2), numsamples=1))
8.6219572235293
>>> next(sample_iter(Normal('',10,2), numsamples=1, library='pymc3'))
array([6.29817056])
```
>Can that not just be handled in the pymc3 code?

Regarding that, I notice the the code for producing samples for pymc3 is different from other libraries. While numpy and scipy use the `do_sample_numpy` and `do_sample_scipy` from sampling directory, pymc3 uses its sample function. 

Why was `do_sample_pymc3` not used although it was made? I will be trying to implement it for now.